src/main.cpp

```diff
@@ -1,48 +1,61 @@
 #include "tools.h"
 
 static bool argumentsHandler(const std::vector<std::string_view>& args)
 {
-	for (const auto& arg : args) {
-		if (arg == "--help") {
-			std::clog << "Usage:\n"
-			             "\n"
-			             "\t--config=$1\t\tAlternate configuration file path.\n"
-			             "\t--ip=$1\t\t\tIP address of the server.\n"
-			             "\t\t\t\tShould be equal to the global IP.\n"
-			             "\t--http-port=$1\tPort for http to listen on.\n"
-			             "\t--game-port=$1\tPort for game server to listen on.\n";
-			return false;
-		} else if (arg == "--version") {
-			printServerVersion();
-			return false;
-		}
-
-		auto tmp = explodeString(arg, "=");
-
-		if (tmp[0] == "--config")
-			ConfigManager::setString(ConfigManager::CONFIG_FILE, tmp[1]);
-		else if (tmp[0] == "--ip")
-			ConfigManager::setString(ConfigManager::IP, tmp[1]);
-		else if (tmp[0] == "--http-port")
-			ConfigManager::setNumber(ConfigManager::HTTP_PORT, std::stoi(tmp[1].data()));
-		else if (tmp[0] == "--game-port")
-			ConfigManager::setNumber(ConfigManager::GAME_PORT, std::stoi(tmp[1].data()));
-	}
-
-	return true;
+    for (const auto& arg : args) {
+        if (arg == "--help") {
+            spdlog::info("Usage:");
+            spdlog::info("  --config=<path>    Alternate configuration file path.");
+            spdlog::info("  --ip=<address>     IP address of the server.");
+            spdlog::info("  --http-port=<port> Port for http to listen on.");
+            spdlog::info("  --game-port=<port> Port for game server to listen on.");
+            return false;
+        } else if (arg == "--version") {
+            printServerVersion();
+            return false;
+        }
+
+        auto tmp = explodeString(arg, "=");
+
+        if (tmp[0] == "--config")
+            ConfigManager::setString(ConfigManager::CONFIG_FILE, tmp[1]);
+        else if (tmp[0] == "--ip")
+            ConfigManager::setString(ConfigManager::IP, tmp[1]);
+        else if (tmp[0] == "--http-port")
+            ConfigManager::setNumber(ConfigManager::HTTP_PORT, std::stoi(tmp[1].data()));
+        else if (tmp[0] == "--game-port")
+            ConfigManager::setNumber(ConfigManager::GAME_PORT, std::stoi(tmp[1].data()));
+    }
+
+    return true;
 }
 
-int main(int argc, const char** argv)
-{
-	std::vector<std::string_view> args(argv, argv + argc);
-	if (!argumentsHandler(args)) {
-		return 1;
-	}
+int main(int argc, const char** argv) {
+    try {
+        std::vector<std::string_view> args(argv, argv + argc);
+        if (!argumentsHandler(args)) {
+            return 1;
+        }
+
+        spdlog::info("Starting server...");
+        startServer();
 
-	startServer();
-	return 0;
+        spdlog::info("Server stopped");
+        return 0;
+    } catch (const std::exception& e) {
+        spdlog::critical("Exception: {}", e.what());
+        return 1;
+    } catch (...) {
+        spdlog::critical("Unknown exception occurred");
+        return 1;
+    }
```

Comment:
I think we should not have this catch-all clause, let the server crash with the default handler

Also, it should not catch any exceptions if it's a debug build, since it should let the debugger catch the exception instead. You can probably use `#ifndef NDEBUG` (undefined in a release build) to control